google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -65,6 +66,14 @@ class ConnectionWorker implements AutoCloseable {
   // Maximum wait time on inflight quota before error out.
   private static long INFLIGHT_QUOTA_MAX_WAIT_TIME_MILLI = 300000;
 
+  /*
+   * Maximum time waiting for request callback before shutting down the connection.
+   *
+   * We will constantly checking how much time we have been waiting for the next request callback
+   * if we wait too much time we will start shutting down the connections and clean up the queues.
+   */
+  private static Duration MAXIMUM_REQUEST_CALLBACK_WAIT_TIME = Duration.ofMinutes(10);
```

Comment:
To keep it safe, let's put a longer time say 30 minutes.

Reply:
Let's use 20 minutes, 30 minutes sound crazy long for a pipeline to stop

Reply:
Right now the user waits forever on the queue, I am just afraid some user does crazy things. I am fine with 20 min.